mindsdb/mindsdb
# Feat/update otel configs

## Description

This PR updates the OpenTelemetry integration to improve configuration flexibility via environment variables. The changes allow users to define separate endpoints for logging, tracing, and metrics, making it easier to integrate with various OTEL Collectors. It also provides enhanced support for customizing attributes and enabling the SDK in local environments for development and testing purposes.

Fixes #<issue_number>

## Type of change


## Verification Process

To ensure the changes are working as expected:

- [ ] **Test Location**: Use a local setup with the provided `docker-compose.yml` or any custom OTEL Collector configuration.
  1. Set different endpoints for `OTEL_OTLP_LOGGING_ENDPOINT`, `OTEL_OTLP_TRACING_ENDPOINT`, and `OTEL_OTLP_METRICS_ENDPOINT` in the environment variables.
  2. Start the application using `docker-compose up`.
  3. Validate that logs, metrics, and traces are correctly sent to their respective endpoints.
  4. Test the `OTEL_EXTRA_ATTRIBUTES` configuration by including additional attributes and ensuring they appear in the OTEL Collector's output.

## Additional Media:

- [ ] Attached screenshots or a Loom video showcasing the configuration changes and their effects on metrics, logging, and tracing outputs.

## Checklist:
